### ISSUE TITLE
Building hyperapp as minified es6 module

### ISSUE THREAD

**Issue by kumarabhishek** [hyperapp/hyperapp#665]
## Problem Statement
Currently Hyperapp is bundled using rollup (generating ES5 library) and uglified using uglify-js. So For those who want to consume Hyperapp as pure ES6 module which works natively on Chrome at present there is no way out

## Solution Proposal
We should use babel-minify that takes away whitespace and do other optimization to generate pure ES6 module that can be consumed directly in chrome. Refer this sample code snippet to better understand the requirement:

__index.html__
```html
<!DOCTYPE html>
<html>
    <head>
        <script type="module" src="./app.js"></script>
    </head>
    <body>
        <div id="app"></div>
    </body>
</html>
```

__app.js__
```js

import { h, app } from './hyperapp.es.js';

const state = {
  count: 0
}

const actions = {
  down: value => state => ({ count: state.count - value }),
  up: value => state => ({ count: state.count + value })
}

const view = (state, actions) =>
  h("div", {}, [
    h("h1", {}, state.count),
    h("button", { onclick: () => actions.down(1) }, "-"),
    h("button", { onclick: () => actions.up(1) }, "+")
  ])

app(state, actions, view, document.getElementById('app'))
```

As seen above I have kept `hyperapp.es.js` in current directly, but once we agree and this PR is merged then in future we can write statement like this:
```
import { h, app } from 'https://cdnjs.cloudflare.com/ajax/libs/hyperapp/1.2.6/hyperapp.es.js';
```

**Comment by jorgebucaran** [comment]
@kumarabhishek Did you know you can import Hyperapp as an ES module directly from rawgit or unpkg?

```html
<head>
  <script type="module">

  import { h, app } from "https://rawgit.com/hyperapp/hyperapp/master/src/index.js"

  // or 

  import { h, app } from "https://unpkg.com/hyperapp?module"

  </script>
</head>
```

**Comment by kumarabhishek** [comment]
True, but that's not optimised. If we use babe-minify the file size is around 3 KB as compared to source index.js size of around 9 KB.

**Comment by jorgebucaran** [comment]
@kumarabhishek Sounds like we just need to distribute a couple of new bundles?

### Currently

dist/hyperapp.js
dist/hyperapp.js.gz
dist/hyperapp.js.map

### Future

dist/hyperapp.js
dist/hyperapp.js.gz
dist/hyperapp.mjs
dist/hyperapp.mjs.map


Like that?

**Comment by kumarabhishek** [comment]
@jorgebucaran Not that much hurdle. Just a single file hyperapp.es.js. No .gz or sourcemap file as debugging will be on real source and not on transpiled ES5 code through sourcemap. I have the fix also in place and can raise PR, but do not have push permission.

**Comment by jorgebucaran** [comment]
@kumarabhishek If I understand what you are proposing correctly, the pure (not-transpiled) source code will be minified and mangled using babel-minify, so I don't understand how to debug without a sourcemap.

>  I have the fix also in place and can raise PR, but do not have push permission.

Yes, create a PR please, that way I can see what you mean more clearly.

**Comment by kumarabhishek** [comment]
@jorgebucaran I am getting error:
```sh
git -c diff.mnemonicprefix=false -c core.quotepath=false -c credential.helper=sourcetree push -v --tags --set-upstream origin refs/heads/kumarabhishek_es_module:refs/heads/kumarabhishek_es_module 
Pushing to git@github.com:hyperapp/hyperapp.git
ERROR: Permission to hyperapp/hyperapp.git denied to kumarabhishek.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Completed with errors, see above
```

**Comment by frenzzy** [comment]
@kumarabhishek you need to fork repo first: https://help.github.com/articles/fork-a-repo/
and maybe this guide will help you: https://github.com/hyperapp/render/blob/master/.github/CONTRIBUTING.md#pull-requests
(just use `hyperapp/hyperapp` repo instead of `hyperapp/render` in examples)

**Comment by kumarabhishek** [comment]
Thanks @frenzzy I did that. You can have a look at https://github.com/hyperapp/hyperapp/pull/666

**Comment by jorgebucaran** [comment]
Closing as wontfix. You can minify Hyperapp without transpiling ES modules in your own dev environment.
